pkg/newRelic/newrelic.go

```diff
@@ -0,0 +1,42 @@
+package newRelic
+
+import (
+	"github.com/newrelic/go-agent/v3/newrelic"
+)
+
+func NewTransaction(nr *newrelic.Transaction) *NewRelicTransaction {
+	return &NewRelicTransaction{
+		nr: nr,
+	}
+}
+
+type NewRelicTransaction struct {
+	nr *newrelic.Transaction
+}
+
+func (txn *NewRelicTransaction) StartGenericSegment(name string) EndSegment {
```

Comment:
Instead of returning this interface, we can directly returning the End function.

package newRelic

import (
	"github.com/newrelic/go-agent/v3/newrelic"
)

func NewTransaction(nr *newrelic.Transaction) *Transaction {
	return &Transaction{
		nr: nr,
	}
}

type Transaction struct {
	nr *newrelic.Transaction
}

func (txn *Transaction) StartGenericSegment(name string) func() {
	if txn == nil {
		return func() {}
	}
	gs := newrelic.Segment{
		Name: name,
	}
	gs.StartTime = txn.nr.StartSegmentNow()
	return gs.End
}